addons/ofxOsc/src/ofxOscReceiver.h

```diff
@@ -1,101 +1,83 @@
- modification, are permitted provided that the following conditions are met:
- * Redistributions of source code must retain the above copyright
- notice, this list of conditions and the following disclaimer.
- * Redistributions in binary form must reproduce the above copyright
- notice, this list of conditions and the following disclaimer in the
- documentation and/or other materials provided with the distribution.
- * Neither the name of the developer nor the
- names of its contributors may be used to endorse or promote products
- derived from this software without specific prior written permission.
- 
- THIS SOFTWARE IS PROVIDED BY DAMIAN STEWART ''AS IS'' AND ANY
- EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
- WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
- DISCLAIMED. IN NO EVENT SHALL DAMIAN STEWART BE LIABLE FOR ANY
- DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
- (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
- LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
- ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
- (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
- SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
- */
-
+// copyright (c) openFrameworks team 2010-2017
+// copyright (c) Damian Stewart 2007-2009
 #pragma once
 
-#include <deque>
-#include "ofMain.h"
-
-#ifdef TARGET_WIN32
-// threads
-#include <windows.h>
-#else
-// threads
-#include <pthread.h>
-#endif
+#include "ofxOscMessage.h"
+#include "ofParameter.h"
+#include "ofThreadChannel.h"
 
-// osc
 #include "OscTypes.h"
 #include "OscPacketListener.h"
 #include "UdpSocket.h"
 
-// ofxOsc
-#include "ofxOscMessage.h"
-
-class ofxOscReceiver : public osc::OscPacketListener
-{
+/// \class ofxOscReceiver
+/// \brief OSC message receiver which listens on a network port
+class ofxOscReceiver : public osc::OscPacketListener {
 public:
+
 	ofxOscReceiver();
 	ofxOscReceiver(const ofxOscReceiver & mom);
-	ofxOscReceiver & operator=(const ofxOscReceiver & mom);
-
-	/// listen_port is the port to listen for messages on
-	bool setup( int listen_port );
-
-	/// returns true if there are any messages waiting for collection
+	ofxOscReceiver& operator=(const ofxOscReceiver & mom);
+	/// for operator= and copy constructor
+	ofxOscReceiver& copy(const ofxOscReceiver& other);
+
+	/// set up the receiver with the port to listen for messages on
+	///
+	/// multiple receivers can share the same port if port reuse is
+	/// enabled (true by default)
+	///
+	/// if port is set to 0, receiver will try to use current port value to
+	/// restart listening
+	///
+	/// if start is true, receiver will start listening (default)
+	/// if start is false, port value is set without starting
+	///
+	/// \return true on success
+	bool setup(int port=0, bool start=true);
```

Comment:
Ok, I split the functionality within the setup() function into two more functions which allows for more manual control without overloading setup(). This also keeps the original functionality of setup() the same for backwards compatibility:

~~~
/// set up the receiver with the port to listen for messages on
	/// and start listening
	///
	/// multiple receivers can share the same port if port reuse is
	/// enabled (true by default)
	///
	/// \return true if listening started successfully
	bool setup(int port);
	
	/// set the receiver port manually
	///
	/// does not *start* listening, restarts receiver if already running
	///
	/// multiple receivers can share the same port if port reuse is
	/// enabled (true by default)
	///
	/// \returns true if port set successfully
	bool setPort(int port);
	
	/// start listening manually using the current port value
	///
	/// note: this is not required if you called setup()
	///
	/// \return true if listening started or was already running
	bool start();
	
	/// stop listening, does not clear port value
	void stop();
~~~

I also renamed clear() to stop() as I think that more accurately fits it's function.